webapp/src/hooks/useApiRequestCompletionState.ts

```diff
@@ -0,0 +1,49 @@
+import {useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {ReduxState} from 'types/common/store.d';
+
+import {resetApiRequestCompletionState} from 'reducers/apiRequest';
+import {msTeamsPluginApi} from 'services';
+
+type Props = {
+    handleSuccess?: () => void
+    handleError?: (error: APIError) => void
+    serviceName: PluginApiServiceName
+    payload?: APIRequestPayload,
+}
+
+function useApiRequestCompletionState({handleSuccess, handleError, serviceName, payload}: Props) {
+    const state = useSelector((reduxState: ReduxState) => reduxState['plugins-com.mattermost.msteams-sync']);
+    const apiState = msTeamsPluginApi.endpoints[serviceName].select(payload)(state);
+
+    const requests = state.apiRequestCompletionSlice.requests;
+    const dispatch = useDispatch();
+
+    // Observe for the change in redux state after API call and do the required actions
+    useEffect(() => {
+        if (
+            requests.includes(serviceName) &&
+            apiState
+        ) {
+            const {isError, isSuccess, isUninitialized, error} = apiState;
+            if (isSuccess && !isError) {
+                handleSuccess?.();
+            }
+
+            if (!isSuccess && isError) {
+                handleError?.(error as APIError);
+            }
+
+            if (!isUninitialized) {
+                dispatch(resetApiRequestCompletionState(serviceName));
+            }
+        }
+    }, [
+        requests.includes(serviceName),
+        apiState,
+        resetApiRequestCompletionState(serviceName),
```

Comment:
I'm not super familiar with Redux Toolkit so I might be incorrect here, but this seems like `resetApiRequestCompletionState(serviceName)` is going to return a new object every time it's called which would cause the `useEffect` to re-run on every render cycle. While the `exhaustive-deps` ESLint rule would probably complain, I feel like we shouldn't include this as a dependency here